static/js/setup-subdomain.js

```diff
@@ -34,6 +35,36 @@ function setupForm() {
     }
   }
 
+  let haveRestartedNtp = false;
+  function pollNtp() {
+    API.getNtpStatus().then((body) => {
+      if (body.statusImplemented) {
+        if (body.synchronized) {
+          ntpMessage.classList.add('hidden');
+        } else {
+          ntpMessage.classList.remove('hidden');
+
+          if (!haveRestartedNtp) {
+            API.restartNtpSync().catch((e) => {
+              console.error('Failed to restart NTP sync:', e);
+            });
+            haveRestartedNtp = true;
+          }
+
+          // poll again in 5 seconds
+          setTimeout(pollNtp, 5000);
```

Comment:
It's a shame we have to poll here because 5 seconds may actually feel like quite a long time, but I can't think of an obvious alternative given how the underlying system service works.

Reply:
I know... I considered making a WebSocket, but the server was just going to be polling anyway.